R/FilterState.R

```diff
@@ -336,42 +368,49 @@ InteractiveFilterState <- R6::R6Class( # nolint
     },
 
     #' @description
-    #' Set state.
+    #' Sets filtering state.
     #'
-    #' @param state (`list`)\cr
-    #'  contains fields relevant for a specific class:
-    #' \itemize{
-    #' \item{`selected`}{ defines initial selection}
-    #' \item{`keep_na` (`logical`)}{ defines whether to keep or remove `NA` values}
-    #' }
+    #' @param state a (`teal_slice`) object; see `Filter state specification`
     #'
     #' @return NULL invisibly
     #'
     set_state = function(state) {
-      logger::log_trace(sprintf(
-        "%s$set_state, dataname: %s setting state of variable %s to: selected=%s, keep_na=%s",
-        class(self)[1],
-        private$dataname,
-        private$varname,
-        paste(state$selected, collapse = " "),
-        state$keep_na
-      ))
-      stopifnot(is.list(state) && all(names(state) %in% c("selected", "keep_na")))
-
-      if (!is.null(state$keep_na)) {
-        self$set_keep_na(state$keep_na)
-      }
-      if (!is.null(state$selected)) {
-        self$set_selected(state$selected)
-      }
-      logger::log_trace(
-        sprintf(
-          "%s$set_state, dataname: %s done setting state for variable %s",
+      if (inherits(state, "teal_slice")) {
+        if (!is.null(state$selected)) {
+          self$set_selected(state$selected)
+        }
+        if (!is.null(state$keep_na)) {
+          self$set_keep_na(state$keep_na)
+        }
+        if (!is.null(state$keep_inf)) {
+          self$set_keep_inf(state$keep_inf)
+        }
```

Comment:
This method is not available in the abstract class I suggest to remove
```suggestion
```

and make in RangeFilterState
```r
      set_state = function(state) {
         super$set_state(state)
         if (!is.null(state$keep_inf)) {
          self$set_keep_inf(state$keep_inf)
        }
      }
```